app/assets/javascripts/controllers/topology/cloud_topology_controller.js

```diff
@@ -31,167 +38,169 @@ function CloudTopologyCtrl($scope, $http, $interval, $location, topologyService,
   function getCloudTopologyFormDataComplete(response) {
     var data = response.data;
 
-    var currentSelectedKinds = $scope.kinds;
+    var currentSelectedKinds = vm.kinds;
 
-    $scope.items = data.data.items;
-    $scope.relations = data.data.relations;
-    $scope.kinds = data.data.kinds;
+    vm.items = data.data.items;
+    vm.relations = data.data.relations;
+    // NOTE: $scope.kind is required by kubernetes-topology-icon
+    vm.kinds = $scope.kinds = data.data.kinds;
     icons = data.data.icons;
 
-    if (currentSelectedKinds && (Object.keys(currentSelectedKinds).length !== Object.keys($scope.kinds).length)) {
-      $scope.kinds = currentSelectedKinds;
+    if (currentSelectedKinds && (Object.keys(currentSelectedKinds).length !== Object.keys(vm.kinds).length)) {
+      vm.kinds = vm.kinds = currentSelectedKinds;
     }
   }
 
-  $scope.checkboxModel = {
-    value: false
+  vm.checkboxModel = {
+    value: false,
   };
 
-  $scope.legendTooltip = __("Click here to show/hide entities of this type");
-
-  $('input#box_display_names').click(topologyService.showHideNames($scope));
-  $scope.refresh();
-  var promise = $interval($scope.refresh, 1000 * 60 * 3);
+  vm.legendTooltip = __('Click here to show/hide entities of this type');
 
+  $('input#box_display_names').click(topologyService.showHideNames(vm));
+  vm.refresh();
+  var promise = $interval(vm.refresh, 1000 * 60 * 3);
   $scope.$on('$destroy', function() {
     $interval.cancel(promise);
   });
 
-  $scope.$on("render", function(ev, vertices, added) {
+  $scope.$on('render', function(ev, vertices, added) {
     /*
      * We are passed two selections of <g> elements:
      * vertices: All the elements
      * added: Just the ones that were added
      */
 
-    added.attr("class", function(d) {
+    added.attr('class', function(d) {
       return d.item.kind;
     });
 
-    added.append("circle")
-      .attr("r", function(d) {
-        return self.getDimensions(d).r;
+    added.append('circle')
+      .attr('r', function(d) {
+        return vm.getDimensions(d).r;
       })
       .attr('class', function(d) {
         return topologyService.getItemStatusClass(d);
       })
-      .on("contextmenu", function(d) {
-        self.contextMenu(this, d);
+      .on('contextmenu', function(d) {
+        vm.contextMenu(this, d);
+        vm.contextMenu(this, d);
```

Comment:
Correct in the sense: do we really need two identical lines of code here? :smile: